extensions/PhabBugz/lib/Feed.pm

```diff
@@ -0,0 +1,223 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::PhabBugz::Feed;
+
+use 5.10.1;
+
+use Moo;
+
+use Bugzilla::Extension::PhabBugz::Constants;
+use Bugzilla::Extension::PhabBugz::Revision;
+use Bugzilla::Extension::PhabBugz::Util qw(
+    add_security_sync_comments
+    create_private_revision_policy
+    create_revision_attachment
+    edit_revision_policy
+    get_bug_role_phids
+    get_phab_bmo_ids
+    get_security_sync_groups
+    is_attachment_phab_revision
+    make_revision_public
+    request
+    set_phab_user
+);
+
+has 'is_daemon' => ( is => 'rw', default => 0 );
+has 'logger'    => ( is => 'rw' );
+
+sub start {
+    my ($self) = @_;
+    while (1) {
+        if (Bugzilla->params->{phabricator_enabled}) {
+            $self->feed_query();
+        }
+        sleep(PHAB_POLL_SECONDS);
+        Bugzilla->_cleanup();
+    }
+}
+
+sub feed_query {
+    my ($self) = @_;
+    my $dbh = Bugzilla->dbh;
+
+    # Ensure Phabricator syncing is enabled
+    if (!Bugzilla->params->{phabricator_enabled}) {
+        $self->logger->info("PHABRICATOR SYNC DISABLED");
+        return;
+    }
+
+    $self->logger->info("FEED: Fetching new transactions");
+
+    my $last_ts = $dbh->selectrow_array("
+        SELECT value FROM phabbugz WHERE name = 'feed_last_ts'");
+    $last_ts ||= 0;
+    $self->logger->debug("LAST_TS: $last_ts");
+
+    # Check for new transctions (stories)
+    my $transactions = $self->feed_transactions($last_ts);
+    if (!%$transactions) {
+        $self->logger->info("FEED: No new transactions");
+        return;
+    }
+
+    # Process each story
+    foreach my $story (keys %$transactions) {
+        my $skip = 0;
+        my $story_data  = $transactions->{$story};
+        my $author_phid = $story_data->{authorPHID};
+        my $object_phid = $story_data->{objectPHID};
+        my $story_text  = $story_data->{text};
+        my $story_epoch = $story_data->{epoch};
+
+        $self->logger->debug("STORY PHID: $story");
+        $self->logger->debug("STORY_EPOCH: $story_epoch");
+        $self->logger->debug("AUTHOR PHID: $author_phid");
+        $self->logger->debug("OBJECT PHID: $object_phid");
+        $self->logger->debug("STORY TEXT: $story_text");
+
+        # Only interested in changes to revisions for now.
+        if ($object_phid !~ /^PHID-DREV/) {
+            $self->logger->debug("SKIP: Not a revision change");
+            $skip = 1;
+        }
+
+        # Skip changes done by phab-bot user
+        my $phab_users = get_phab_bmo_ids({ phids => [$author_phid] });
+        if (@$phab_users) {
```

Comment:
Do we want to avoid this `if()` if `$skip` is already flagged?

Reply:
If `get_phab_bmo_ids` is expensive, then yes for certain.